engine/lite/request-response/runtime/src/test/scala/pl/touk/nussknacker/engine/requestresponse/test/RunnableEspScenario.scala

```diff
@@ -0,0 +1,81 @@
+package pl.touk.nussknacker.engine.requestresponse.test
+
+import cats.data.ValidatedNel
+import com.typesafe.config.ConfigFactory
+import io.dropwizard.metrics5.MetricRegistry
+import org.scalatest.Matchers
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.process.ComponentUseCase
+import pl.touk.nussknacker.engine.api.runtimecontext.IncContextIdGenerator
+import pl.touk.nussknacker.engine.api.test.TestData
+import pl.touk.nussknacker.engine.graph.EspProcess
+import pl.touk.nussknacker.engine.lite.api.commonTypes.ErrorType
+import pl.touk.nussknacker.engine.lite.api.runtimecontext.LiteEngineRuntimeContextPreparer
+import pl.touk.nussknacker.engine.lite.metrics.dropwizard.DropwizardMetricsProviderFactory
+import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+import pl.touk.nussknacker.engine.requestresponse.{FutureBasedRequestResponseScenarioInterpreter, RequestResponseConfigCreator, RequestResponseInterpreter}
+import pl.touk.nussknacker.engine.requestresponse.metrics.InvocationMetrics
+import pl.touk.nussknacker.engine.resultcollector.ProductionServiceInvocationCollector
+import pl.touk.nussknacker.engine.testing.LocalModelData
+import pl.touk.nussknacker.engine.testmode.TestProcess.TestResults
+import pl.touk.nussknacker.test.PatientScalaFutures
+
+import java.nio.charset.StandardCharsets
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Using
+
+trait RunnableEspScenario extends Matchers with PatientScalaFutures {
```

Comment:
Have u taken a look into `LiteTestScenarioRunnerSpec` and `RequestResponseTestScenarioRunner` in our brand new `lite-components-testkit` module? We have plans to have one, standarized, well documented way to test components by running scenarios. We planned to make it quite runtime agnostic - no matter if you run your test on flink, streaming-lite or req-resp. We wanted to make it such easy as writing tests for akka http resources, spring boot integration test and so on. We've just started a work on that in previous release but maybe you could take a look into it and eventually extend it if you see sth worth adding?

Reply:
That is nice, looks like something I wanted to created here but it is more generic - I haven't seen that. Do you plan to keep tests for component like `for-each` or `collector` inside `lite-components-testkit` or use it as test dependency? I am asking since `liteComponentsTestkit` depends on `requestResponseRuntime` so project `requestResponseRuntime` cannot depend on dependency that depends on it - or am I missing something?

Reply:
The plan is that testkit is a dependency that will be added to projects with coponents. testkit depnends on runtime because it need runtime to execute scenario :) All components like `fore-each` etc. are in separate modules which are not dependants of runtime so I guess that everything should work as expected?